CONTRIBUTING.md

```diff
@@ -8,13 +8,60 @@ We use Node in this project. To manage the language runtime we recommend using [
 
 With `nvm` use `nvm use` to set the correct node version in your machine.
 
+To setup the Android environment install Android studio. Once installed, use the following commands to setup the android sdk
+
+```
+echo "export ANDROID_HOME=\"$HOME/Library/Android/sdk\"" >> ~/.bash_profile
+echo "export ANDROID_SDK_ROOT=\"$HOME/Library/Android/sdk\"" >> ~/.bash_profile
+source ~/.bash_profile
+```
+
+Install system images
+```
+$ANDROID_SDK_ROOT/tools/bin/sdkmanager "system-images;android-21;google_apis;x86"
+```
+
 ## Setup
 
 Given that you have a compatible environment as stated above you can now setup the project.
 
 + `npm install` to install npm dependencies
 +  Open the project in Android studio
 
+## Testing
+
+#### Snapshot testing
+Create an AVD using the following commands
+
+```
+$ANDROID_HOME/tools/bin/avdmanager create avd --name "bpk-droid-avd" --force --package "system-images;android-21;google_apis;x86" --device "Nexus 4" && cp bpk-droid-local.ini ~/.android/avd/bpk-droid-avd.avd/config.ini
+```
+
+To start the emulator and attach and sdcard to it, run
+ 
+```$ANDROID_HOME/tools/mksdcard -l e 512M sd.img
```

Comment:
The `mksdcard` can be in the "create" section above as it is required only the first time